2025-03-29T04:50:10.8986787Z Requested labels: ubuntu-latest
2025-03-29T04:50:10.8987961Z Job defined at: microsoft/markitdown/.github/workflows/tests.yml@refs/pull/1161/merge
2025-03-29T04:50:10.8988184Z Waiting for a runner to pick up this job...
2025-03-29T04:50:13.5518723Z Job is waiting for a hosted runner to come online.
2025-03-29T04:50:17.7204008Z Job is about to start running on the hosted runner: GitHub Actions 961 (hosted)
2025-03-29T04:50:20.3950269Z Current runner version: '2.323.0'
2025-03-29T04:50:20.3974406Z ##[group]Operating System
2025-03-29T04:50:20.3975195Z Ubuntu
2025-03-29T04:50:20.3975765Z 24.04.2
2025-03-29T04:50:20.3976293Z LTS
2025-03-29T04:50:20.3977030Z ##[endgroup]
2025-03-29T04:50:20.3977792Z ##[group]Runner Image
2025-03-29T04:50:20.3978428Z Image: ubuntu-24.04
2025-03-29T04:50:20.3978976Z Version: 20250323.1.0
2025-03-29T04:50:20.3980109Z Included Software: https://github.com/actions/runner-images/blob/ubuntu24/20250323.1/images/ubuntu/Ubuntu2404-Readme.md
2025-03-29T04:50:20.3981527Z Image Release: https://github.com/actions/runner-images/releases/tag/ubuntu24%2F20250323.1
2025-03-29T04:50:20.3982459Z ##[endgroup]
2025-03-29T04:50:20.3983064Z ##[group]Runner Image Provisioner
2025-03-29T04:50:20.3983675Z 2.0.422.1
2025-03-29T04:50:20.3984172Z ##[endgroup]
2025-03-29T04:50:20.3985317Z ##[group]GITHUB_TOKEN Permissions
2025-03-29T04:50:20.3987774Z Contents: read
2025-03-29T04:50:20.3988438Z Metadata: read
2025-03-29T04:50:20.3989148Z Packages: read
2025-03-29T04:50:20.3989831Z ##[endgroup]
2025-03-29T04:50:20.3992912Z Secret source: None
2025-03-29T04:50:20.3993978Z Prepare workflow directory
2025-03-29T04:50:20.4300687Z Prepare all required actions
2025-03-29T04:50:20.4337038Z Getting action download info
2025-03-29T04:50:20.6650462Z ##[group]Download immutable action package 'actions/checkout@v4'
2025-03-29T04:50:20.6651365Z Version: 4.2.2
2025-03-29T04:50:20.6652196Z Digest: sha256:ccb2698953eaebd21c7bf6268a94f9c26518a7e38e27e0b83c1fe1ad049819b1
2025-03-29T04:50:20.6653231Z Source commit SHA: 11bd71901bbe5b1630ceea73d27597364c9af683
2025-03-29T04:50:20.6653863Z ##[endgroup]
2025-03-29T04:50:20.7490337Z ##[group]Download immutable action package 'actions/setup-python@v5'
2025-03-29T04:50:20.7491061Z Version: 5.5.0
2025-03-29T04:50:20.7491730Z Digest: sha256:1fac95d751afd58314aebd2ab01d4bf3b3109a441917bd2cac2a22898ec494f7
2025-03-29T04:50:20.7492615Z Source commit SHA: 8d9ed9ac5c53483de85588cdf95a591a75ab9f55
2025-03-29T04:50:20.7493252Z ##[endgroup]
2025-03-29T04:50:21.0910995Z Complete job name: tests
2025-03-29T04:50:21.1616400Z ##[group]Run actions/checkout@v4
2025-03-29T04:50:21.1617602Z with:
2025-03-29T04:50:21.1618031Z   repository: microsoft/markitdown
2025-03-29T04:50:21.1618701Z   token: ***
2025-03-29T04:50:21.1619102Z   ssh-strict: true
2025-03-29T04:50:21.1619513Z   ssh-user: git
2025-03-29T04:50:21.1619934Z   persist-credentials: true
2025-03-29T04:50:21.1620398Z   clean: true
2025-03-29T04:50:21.1620820Z   sparse-checkout-cone-mode: true
2025-03-29T04:50:21.1621320Z   fetch-depth: 1
2025-03-29T04:50:21.1621776Z   fetch-tags: false
2025-03-29T04:50:21.1622194Z   show-progress: true
2025-03-29T04:50:21.1622608Z   lfs: false
2025-03-29T04:50:21.1622997Z   submodules: false
2025-03-29T04:50:21.1623420Z   set-safe-directory: true
2025-03-29T04:50:21.1624091Z ##[endgroup]
2025-03-29T04:50:21.3256278Z Syncing repository: microsoft/markitdown
2025-03-29T04:50:21.3258272Z ##[group]Getting Git version info
2025-03-29T04:50:21.3259007Z Working directory is '/home/runner/work/markitdown/markitdown'
2025-03-29T04:50:21.3260050Z [command]/usr/bin/git version
2025-03-29T04:50:21.3304812Z git version 2.49.0
2025-03-29T04:50:21.3332347Z ##[endgroup]
2025-03-29T04:50:21.3345636Z Temporarily overriding HOME='/home/runner/work/_temp/56970e23-6f99-4169-88ff-3675fac30df5' before making global git config changes
2025-03-29T04:50:21.3347260Z Adding repository directory to the temporary git global config as a safe directory
2025-03-29T04:50:21.3350919Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/markitdown/markitdown
2025-03-29T04:50:21.3389386Z Deleting the contents of '/home/runner/work/markitdown/markitdown'
2025-03-29T04:50:21.3393532Z ##[group]Initializing the repository
2025-03-29T04:50:21.3398835Z [command]/usr/bin/git init /home/runner/work/markitdown/markitdown
2025-03-29T04:50:21.3463163Z hint: Using 'master' as the name for the initial branch. This default branch name
2025-03-29T04:50:21.3464553Z hint: is subject to change. To configure the initial branch name to use in all
2025-03-29T04:50:21.3465755Z hint: of your new repositories, which will suppress this warning, call:
2025-03-29T04:50:21.3466428Z hint:
2025-03-29T04:50:21.3467268Z hint: 	git config --global init.defaultBranch <name>
2025-03-29T04:50:21.3468458Z hint:
2025-03-29T04:50:21.3469053Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2025-03-29T04:50:21.3470646Z hint: 'development'. The just-created branch can be renamed via this command:
2025-03-29T04:50:21.3471984Z hint:
2025-03-29T04:50:21.3472710Z hint: 	git branch -m <name>
2025-03-29T04:50:21.3474041Z Initialized empty Git repository in /home/runner/work/markitdown/markitdown/.git/
2025-03-29T04:50:21.3481558Z [command]/usr/bin/git remote add origin https://github.com/microsoft/markitdown
2025-03-29T04:50:21.3513472Z ##[endgroup]
2025-03-29T04:50:21.3514676Z ##[group]Disabling automatic garbage collection
2025-03-29T04:50:21.3518605Z [command]/usr/bin/git config --local gc.auto 0
2025-03-29T04:50:21.3547405Z ##[endgroup]
2025-03-29T04:50:21.3548671Z ##[group]Setting up auth
2025-03-29T04:50:21.3554867Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-03-29T04:50:21.3585804Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-03-29T04:50:21.3880278Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2025-03-29T04:50:21.3910929Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2025-03-29T04:50:21.4132692Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2025-03-29T04:50:21.4172765Z ##[endgroup]
2025-03-29T04:50:21.4174032Z ##[group]Fetching the repository
2025-03-29T04:50:21.4183375Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +d2eee798af4c31ead46452b3db57edd87b05940d:refs/remotes/pull/1161/merge
2025-03-29T04:50:22.1344540Z From https://github.com/microsoft/markitdown
2025-03-29T04:50:22.1345732Z  * [new ref]         d2eee798af4c31ead46452b3db57edd87b05940d -> pull/1161/merge
2025-03-29T04:50:22.1371089Z ##[endgroup]
2025-03-29T04:50:22.1371895Z ##[group]Determining the checkout info
2025-03-29T04:50:22.1373147Z ##[endgroup]
2025-03-29T04:50:22.1378178Z [command]/usr/bin/git sparse-checkout disable
2025-03-29T04:50:22.1415265Z [command]/usr/bin/git config --local --unset-all extensions.worktreeConfig
2025-03-29T04:50:22.1442440Z ##[group]Checking out the ref
2025-03-29T04:50:22.1445827Z [command]/usr/bin/git checkout --progress --force refs/remotes/pull/1161/merge
2025-03-29T04:50:22.1688002Z Note: switching to 'refs/remotes/pull/1161/merge'.
2025-03-29T04:50:22.1689010Z 
2025-03-29T04:50:22.1689651Z You are in 'detached HEAD' state. You can look around, make experimental
2025-03-29T04:50:22.1690814Z changes and commit them, and you can discard any commits you make in this
2025-03-29T04:50:22.1691988Z state without impacting any branches by switching back to a branch.
2025-03-29T04:50:22.1693240Z 
2025-03-29T04:50:22.1694172Z If you want to create a new branch to retain commits you create, you may
2025-03-29T04:50:22.1695903Z do so (now or later) by using -c with the switch command. Example:
2025-03-29T04:50:22.1697211Z 
2025-03-29T04:50:22.1697640Z   git switch -c <new-branch-name>
2025-03-29T04:50:22.1698331Z 
2025-03-29T04:50:22.1698716Z Or undo this operation with:
2025-03-29T04:50:22.1699313Z 
2025-03-29T04:50:22.1699668Z   git switch -
2025-03-29T04:50:22.1700142Z 
2025-03-29T04:50:22.1700916Z Turn off this advice by setting config variable advice.detachedHead to false
2025-03-29T04:50:22.1702114Z 
2025-03-29T04:50:22.1703529Z HEAD is now at d2eee79 Merge b20861442fafb0e46dd0363f08ea88df535968aa into 3fcd48cdfc651cbf508071c8d2fb7d82aeb075de
2025-03-29T04:50:22.1707543Z ##[endgroup]
2025-03-29T04:50:22.1735375Z [command]/usr/bin/git log -1 --format=%H
2025-03-29T04:50:22.1757410Z d2eee798af4c31ead46452b3db57edd87b05940d
2025-03-29T04:50:22.2010253Z ##[group]Run actions/setup-python@v5
2025-03-29T04:50:22.2010868Z with:
2025-03-29T04:50:22.2011313Z   python-version: 3.10
3.11
3.12

2025-03-29T04:50:22.2011852Z   check-latest: false
2025-03-29T04:50:22.2012473Z   token: ***
2025-03-29T04:50:22.2012914Z   update-environment: true
2025-03-29T04:50:22.2013423Z   allow-prereleases: false
2025-03-29T04:50:22.2013930Z   freethreaded: false
2025-03-29T04:50:22.2014390Z ##[endgroup]
2025-03-29T04:50:22.3639504Z ##[group]Installed versions
2025-03-29T04:50:22.3701697Z Successfully set up CPython (3.10.16)
2025-03-29T04:50:22.3718624Z Successfully set up CPython (3.11.11)
2025-03-29T04:50:22.3730510Z Successfully set up CPython (3.12.9)
2025-03-29T04:50:22.3732888Z ##[endgroup]
2025-03-29T04:50:22.3871596Z ##[group]Run pipx install hatch
2025-03-29T04:50:22.3872708Z [36;1mpipx install hatch[0m
2025-03-29T04:50:22.3928387Z shell: /usr/bin/bash -e {0}
2025-03-29T04:50:22.3929327Z env:
2025-03-29T04:50:22.3930295Z   pythonLocation: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:22.3932018Z   PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib/pkgconfig
2025-03-29T04:50:22.3933722Z   Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:22.3935240Z   Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:22.3936915Z   Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:22.3939895Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib:/opt/hostedtoolcache/Python/3.11.11/x64/lib:/opt/hostedtoolcache/Python/3.10.16/x64/lib
2025-03-29T04:50:22.3942626Z ##[endgroup]
2025-03-29T04:50:22.5352178Z creating virtual environment...
2025-03-29T04:50:22.6261559Z installing hatch...
2025-03-29T04:50:27.8445112Z done! ✨ 🌟 ✨
2025-03-29T04:50:27.8447477Z   installed package hatch 1.14.0, installed using Python 3.12.3
2025-03-29T04:50:27.8448497Z   These apps are now globally available
2025-03-29T04:50:27.8449216Z     - hatch
2025-03-29T04:50:27.8654579Z ##[group]Run cd packages/markitdown; hatch test
2025-03-29T04:50:27.8655032Z [36;1mcd packages/markitdown; hatch test[0m
2025-03-29T04:50:27.8705116Z shell: /usr/bin/bash -e {0}
2025-03-29T04:50:27.8705412Z env:
2025-03-29T04:50:27.8705716Z   pythonLocation: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:27.8706174Z   PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib/pkgconfig
2025-03-29T04:50:27.8706810Z   Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:27.8707300Z   Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:27.8707706Z   Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
2025-03-29T04:50:27.8708396Z   LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib:/opt/hostedtoolcache/Python/3.11.11/x64/lib:/opt/hostedtoolcache/Python/3.10.16/x64/lib
2025-03-29T04:50:27.8709057Z ##[endgroup]
2025-03-29T04:50:28.0975752Z Creating environment: hatch-test.py3.12
2025-03-29T04:50:28.1460429Z Installing project in development mode
2025-03-29T04:50:30.0734353Z Checking dependencies
2025-03-29T04:50:30.1183336Z Syncing dependencies
2025-03-29T04:50:34.7526033Z ============================= test session starts ==============================
2025-03-29T04:50:34.7526517Z platform linux -- Python 3.12.3, pytest-8.3.5, pluggy-1.5.0
2025-03-29T04:50:34.7527151Z rootdir: /home/runner/work/markitdown/markitdown/packages/markitdown
2025-03-29T04:50:34.7527553Z configfile: pyproject.toml
2025-03-29T04:50:34.7527891Z plugins: rerunfailures-14.0, anyio-4.9.0, xdist-3.6.1, mock-3.14.0
2025-03-29T04:50:34.7528231Z collected 172 items
2025-03-29T04:50:34.7528377Z 
2025-03-29T04:50:36.4759750Z tests/test_cli_misc.py ..                                                [  1%]
2025-03-29T04:51:16.4467723Z tests/test_cli_vectors.py ...F...........F...........F........ssssssssss [ 27%]
2025-03-29T04:51:19.8081201Z ss.F                                                                     [ 30%]
2025-03-29T04:51:20.1365360Z tests/test_module_misc.py ......ss.ss                                    [ 36%]
2025-03-29T04:51:26.5767879Z tests/test_module_vectors.py ..................F..............F......... [ 61%]
2025-03-29T04:51:40.3167368Z .....F...........sssssssssssssss...F..............F............F.F       [100%]
2025-03-29T04:51:40.3167839Z 
2025-03-29T04:51:40.3168016Z =================================== FAILURES ===================================
2025-03-29T04:51:40.3168569Z _____________________ test_output_to_stdout[test_vector3] ______________________
2025-03-29T04:51:40.3169025Z 
2025-03-29T04:51:40.3169349Z shared_tmp_dir = PosixPath('/tmp/pytest-of-runner/pytest-0/pytest_tmp0')
2025-03-29T04:51:40.3171356Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3173057Z 
2025-03-29T04:51:40.3175113Z     @pytest.mark.parametrize("test_vector", CLI_TEST_VECTORS)
2025-03-29T04:51:40.3177567Z     def test_output_to_stdout(shared_tmp_dir, test_vector) -> None:
2025-03-29T04:51:40.3178263Z         """Test that the CLI outputs to stdout correctly."""
2025-03-29T04:51:40.3178770Z     
2025-03-29T04:51:40.3179099Z         result = subprocess.run(
2025-03-29T04:51:40.3179529Z             [
2025-03-29T04:51:40.3180385Z                 "python",
2025-03-29T04:51:40.3180817Z                 "-m",
2025-03-29T04:51:40.3181165Z                 "markitdown",
2025-03-29T04:51:40.3181649Z                 os.path.join(TEST_FILES_DIR, test_vector.filename),
2025-03-29T04:51:40.3182168Z             ],
2025-03-29T04:51:40.3182522Z             capture_output=True,
2025-03-29T04:51:40.3182949Z             text=True,
2025-03-29T04:51:40.3183301Z         )
2025-03-29T04:51:40.3183597Z     
2025-03-29T04:51:40.3184107Z >       assert result.returncode == 0, f"CLI exited with error: {result.stderr}"
2025-03-29T04:51:40.3186411Z E       AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3188737Z E           warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3189544Z E         Traceback (most recent call last):
2025-03-29T04:51:40.3190133Z E           File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3190712Z E           File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3191750Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3193019Z E             main()
2025-03-29T04:51:40.3193902Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3194901Z E             result = markitdown.convert(
2025-03-29T04:51:40.3195366Z E                      ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3196357Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3197732Z E             return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3198389Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3199451Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3200693Z E             return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3201694Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3202755Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3203960Z E             raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3204910Z E         markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3205921Z E          - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3206791Z E         
2025-03-29T04:51:40.3207114Z E         
2025-03-29T04:51:40.3207427Z E       assert 1 == 0
2025-03-29T04:51:40.3209170Z E        +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '/home/runner/work/markitdown/markitdown/packages/markitdown/test...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3210817Z 
2025-03-29T04:51:40.3211030Z tests/test_cli_vectors.py:65: AssertionError
2025-03-29T04:51:40.3211674Z ______________________ test_output_to_file[test_vector3] _______________________
2025-03-29T04:51:40.3212124Z 
2025-03-29T04:51:40.3212465Z shared_tmp_dir = PosixPath('/tmp/pytest-of-runner/pytest-0/pytest_tmp0')
2025-03-29T04:51:40.3214340Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3215846Z 
2025-03-29T04:51:40.3216109Z     @pytest.mark.parametrize("test_vector", CLI_TEST_VECTORS)
2025-03-29T04:51:40.3216973Z     def test_output_to_file(shared_tmp_dir, test_vector) -> None:
2025-03-29T04:51:40.3217610Z         """Test that the CLI outputs to a file correctly."""
2025-03-29T04:51:40.3218126Z     
2025-03-29T04:51:40.3218597Z         output_file = os.path.join(shared_tmp_dir, test_vector.filename + ".output")
2025-03-29T04:51:40.3219273Z         result = subprocess.run(
2025-03-29T04:51:40.3219671Z             [
2025-03-29T04:51:40.3220239Z                 "python",
2025-03-29T04:51:40.3220625Z                 "-m",
2025-03-29T04:51:40.3220973Z                 "markitdown",
2025-03-29T04:51:40.3221341Z                 "-o",
2025-03-29T04:51:40.3221691Z                 output_file,
2025-03-29T04:51:40.3222182Z                 os.path.join(TEST_FILES_DIR, test_vector.filename),
2025-03-29T04:51:40.3222695Z             ],
2025-03-29T04:51:40.3223027Z             capture_output=True,
2025-03-29T04:51:40.3223401Z             text=True,
2025-03-29T04:51:40.3223749Z         )
2025-03-29T04:51:40.3224099Z     
2025-03-29T04:51:40.3224571Z >       assert result.returncode == 0, f"CLI exited with error: {result.stderr}"
2025-03-29T04:51:40.3226324Z E       AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3228381Z E           warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3229053Z E         Traceback (most recent call last):
2025-03-29T04:51:40.3229530Z E           File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3230000Z E           File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3230866Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3231681Z E             main()
2025-03-29T04:51:40.3232403Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3233230Z E             result = markitdown.convert(
2025-03-29T04:51:40.3233620Z E                      ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3234638Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3235625Z E             return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3236196Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3237289Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3238325Z E             return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3238889Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3239748Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3240682Z E             raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3241454Z E         markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3242358Z E          - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3242946Z E         
2025-03-29T04:51:40.3243192Z E         
2025-03-29T04:51:40.3243438Z E       assert 1 == 0
2025-03-29T04:51:40.3244764Z E        +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '-o', '/tmp/pytest-of-runner/pytest-0/pytest_tmp0/test.pptx.outpu...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3246045Z 
2025-03-29T04:51:40.3246208Z tests/test_cli_vectors.py:90: AssertionError
2025-03-29T04:51:40.3246917Z ______________ test_input_from_stdin_without_hints[test_vector3] _______________
2025-03-29T04:51:40.3247318Z 
2025-03-29T04:51:40.3247593Z shared_tmp_dir = PosixPath('/tmp/pytest-of-runner/pytest-0/pytest_tmp0')
2025-03-29T04:51:40.3249420Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3250795Z 
2025-03-29T04:51:40.3251039Z     @pytest.mark.parametrize("test_vector", CLI_TEST_VECTORS)
2025-03-29T04:51:40.3251675Z     def test_input_from_stdin_without_hints(shared_tmp_dir, test_vector) -> None:
2025-03-29T04:51:40.3252272Z         """Test that the CLI readds from stdin correctly."""
2025-03-29T04:51:40.3252675Z     
2025-03-29T04:51:40.3252930Z         test_input = b""
2025-03-29T04:51:40.3253417Z         with open(os.path.join(TEST_FILES_DIR, test_vector.filename), "rb") as stream:
2025-03-29T04:51:40.3253968Z             test_input = stream.read()
2025-03-29T04:51:40.3254303Z     
2025-03-29T04:51:40.3254564Z         result = subprocess.run(
2025-03-29T04:51:40.3254903Z             [
2025-03-29T04:51:40.3255167Z                 "python",
2025-03-29T04:51:40.3255467Z                 "-m",
2025-03-29T04:51:40.3255764Z                 "markitdown",
2025-03-29T04:51:40.3256177Z                 os.path.join(TEST_FILES_DIR, test_vector.filename),
2025-03-29T04:51:40.3256587Z             ],
2025-03-29T04:51:40.3257013Z             input=test_input,
2025-03-29T04:51:40.3257341Z             capture_output=True,
2025-03-29T04:51:40.3257671Z             text=False,
2025-03-29T04:51:40.3257960Z         )
2025-03-29T04:51:40.3258201Z     
2025-03-29T04:51:40.3258581Z         stdout = result.stdout.decode(locale.getpreferredencoding())
2025-03-29T04:51:40.3259093Z >       assert (
2025-03-29T04:51:40.3259373Z             result.returncode == 0
2025-03-29T04:51:40.3259810Z         ), f"CLI exited with error: {result.stderr.decode('utf-8')}"
2025-03-29T04:51:40.3261455Z E       AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3263433Z E           warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3264091Z E         Traceback (most recent call last):
2025-03-29T04:51:40.3264569Z E           File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3265049Z E           File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3265908Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3266883Z E             main()
2025-03-29T04:51:40.3267605Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3268433Z E             result = markitdown.convert(
2025-03-29T04:51:40.3268825Z E                      ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3269646Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3270625Z E             return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3271169Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3272057Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3273090Z E             return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3273649Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3274498Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3275418Z E             raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3276360Z E         markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3277443Z E          - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3278038Z E         
2025-03-29T04:51:40.3278282Z E         
2025-03-29T04:51:40.3278533Z E       assert 1 == 0
2025-03-29T04:51:40.3279951Z E        +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '/home/runner/work/markitdown/markitdown/packages/markitdown/test...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3281332Z 
2025-03-29T04:51:40.3281503Z tests/test_cli_vectors.py:125: AssertionError
2025-03-29T04:51:40.3282047Z _______________ test_output_to_file_with_data_uris[test_vector1] _______________
2025-03-29T04:51:40.3282459Z 
2025-03-29T04:51:40.3282749Z shared_tmp_dir = PosixPath('/tmp/pytest-of-runner/pytest-0/pytest_tmp0')
2025-03-29T04:51:40.3284387Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati.../jpeg;base64,/9j/4AAQSkZJRgABAQE'], must_not_include=['![This phrase of the caption is Human-written.](Picture4.jpg)'])
2025-03-29T04:51:40.3285791Z 
2025-03-29T04:51:40.3286062Z     @pytest.mark.parametrize("test_vector", DATA_URI_TEST_VECTORS)
2025-03-29T04:51:40.3286865Z     def test_output_to_file_with_data_uris(shared_tmp_dir, test_vector) -> None:
2025-03-29T04:51:40.3287552Z         """Test CLI functionality when keep_data_uris is enabled"""
2025-03-29T04:51:40.3288001Z     
2025-03-29T04:51:40.3288409Z         output_file = os.path.join(shared_tmp_dir, test_vector.filename + ".output")
2025-03-29T04:51:40.3288968Z         result = subprocess.run(
2025-03-29T04:51:40.3289300Z             [
2025-03-29T04:51:40.3289560Z                 "python",
2025-03-29T04:51:40.3290037Z                 "-m",
2025-03-29T04:51:40.3290316Z                 "markitdown",
2025-03-29T04:51:40.3290650Z                 "--keep-data-uris",
2025-03-29T04:51:40.3290986Z                 "-o",
2025-03-29T04:51:40.3291274Z                 output_file,
2025-03-29T04:51:40.3291646Z                 os.path.join(TEST_FILES_DIR, test_vector.filename),
2025-03-29T04:51:40.3292071Z             ],
2025-03-29T04:51:40.3292356Z             capture_output=True,
2025-03-29T04:51:40.3292683Z             text=True,
2025-03-29T04:51:40.3292950Z         )
2025-03-29T04:51:40.3293182Z     
2025-03-29T04:51:40.3293584Z >       assert result.returncode == 0, f"CLI exited with error: {result.stderr}"
2025-03-29T04:51:40.3295224Z E       AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3297100Z E           warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3297755Z E         Traceback (most recent call last):
2025-03-29T04:51:40.3298205Z E           File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3298680Z E           File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3299518Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3300313Z E             main()
2025-03-29T04:51:40.3301021Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3301821Z E             result = markitdown.convert(
2025-03-29T04:51:40.3302192Z E                      ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3302982Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3304004Z E             return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3304693Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3305590Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3306738Z E             return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3307320Z E                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3308182Z E           File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3309116Z E             raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3309882Z E         markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3310774Z E          - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3311349Z E         
2025-03-29T04:51:40.3311594Z E         
2025-03-29T04:51:40.3311835Z E       assert 1 == 0
2025-03-29T04:51:40.3313118Z E        +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '--keep-data-uris', '-o', '/tmp/pytest-of-runner/pytest-0/pytest_...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3314341Z 
2025-03-29T04:51:40.3314504Z tests/test_cli_vectors.py:179: AssertionError
2025-03-29T04:51:40.3315032Z _______________________ test_convert_local[test_vector3] _______________________
2025-03-29T04:51:40.3315398Z 
2025-03-29T04:51:40.3316592Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3318251Z 
2025-03-29T04:51:40.3318506Z     @pytest.mark.parametrize("test_vector", GENERAL_TEST_VECTORS)
2025-03-29T04:51:40.3319020Z     def test_convert_local(test_vector):
2025-03-29T04:51:40.3319426Z         """Test the conversion of a local file."""
2025-03-29T04:51:40.3319831Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3320155Z     
2025-03-29T04:51:40.3320406Z >       result = markitdown.convert(
2025-03-29T04:51:40.3320902Z             os.path.join(TEST_FILES_DIR, test_vector.filename), url=test_vector.url
2025-03-29T04:51:40.3321395Z         )
2025-03-29T04:51:40.3321540Z 
2025-03-29T04:51:40.3321670Z tests/test_module_vectors.py:65: 
2025-03-29T04:51:40.3322064Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3322521Z src/markitdown/_markitdown.py:271: in convert
2025-03-29T04:51:40.3323043Z     return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3323614Z src/markitdown/_markitdown.py:325: in convert_local
2025-03-29T04:51:40.3324198Z     return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3324787Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3325074Z 
2025-03-29T04:51:40.3325327Z self = <markitdown._markitdown.MarkItDown object at 0x7f42859f16d0>
2025-03-29T04:51:40.3326382Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3328546Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', c....pptx', local_path='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx', url=None)]
2025-03-29T04:51:40.3330089Z kwargs = {}, res = None
2025-03-29T04:51:40.3330663Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f42859081d0>]
2025-03-29T04:51:40.3332636Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f4286bebef0>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f4286beb8f0>, priority=0.0), ...]
2025-03-29T04:51:40.3334175Z cur_pos = 0
2025-03-29T04:51:40.3365176Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3365840Z 
2025-03-29T04:51:40.3365973Z     def _convert(
2025-03-29T04:51:40.3366435Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3367186Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3367634Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3368033Z     
2025-03-29T04:51:40.3368358Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3368856Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3369277Z     
2025-03-29T04:51:40.3369648Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3370367Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3371250Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3372055Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3372551Z     
2025-03-29T04:51:40.3372931Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3373428Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3373765Z     
2025-03-29T04:51:40.3374100Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3374613Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3375337Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3375883Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3376359Z                 assert (
2025-03-29T04:51:40.3376889Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3377405Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3377880Z     
2025-03-29T04:51:40.3378176Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3378556Z     
2025-03-29T04:51:40.3378837Z                 # Copy any additional global options
2025-03-29T04:51:40.3379337Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3379829Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3380199Z     
2025-03-29T04:51:40.3380541Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3381038Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3381402Z     
2025-03-29T04:51:40.3381744Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3382235Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3382603Z     
2025-03-29T04:51:40.3382988Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3383529Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3383912Z     
2025-03-29T04:51:40.3384215Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3384690Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3385086Z     
2025-03-29T04:51:40.3385333Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3385685Z                 if stream_info is not None:
2025-03-29T04:51:40.3386089Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3386544Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3387123Z     
2025-03-29T04:51:40.3387587Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3388007Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3388372Z     
2025-03-29T04:51:40.3388772Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3389284Z                 _accepts = False
2025-03-29T04:51:40.3389605Z                 try:
2025-03-29T04:51:40.3390024Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3390545Z                 except NotImplementedError:
2025-03-29T04:51:40.3390910Z                     pass
2025-03-29T04:51:40.3391183Z     
2025-03-29T04:51:40.3391533Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3391978Z                 assert (
2025-03-29T04:51:40.3392320Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3392890Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3393413Z     
2025-03-29T04:51:40.3393678Z                 # Attempt the conversion
2025-03-29T04:51:40.3394041Z                 if _accepts:
2025-03-29T04:51:40.3394361Z                     try:
2025-03-29T04:51:40.3394767Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3395249Z                     except Exception:
2025-03-29T04:51:40.3395634Z                         failed_attempts.append(
2025-03-29T04:51:40.3396046Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3396478Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3397050Z                             )
2025-03-29T04:51:40.3397360Z                         )
2025-03-29T04:51:40.3397640Z                     finally:
2025-03-29T04:51:40.3398146Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3398506Z     
2025-03-29T04:51:40.3398757Z                 if res is not None:
2025-03-29T04:51:40.3399125Z                     # Normalize the content
2025-03-29T04:51:40.3399508Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3400006Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3400487Z                     )
2025-03-29T04:51:40.3400892Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3401356Z                     return res
2025-03-29T04:51:40.3401654Z     
2025-03-29T04:51:40.3401984Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3402440Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3402888Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3403617Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3404549Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3405007Z 
2025-03-29T04:51:40.3405225Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3405804Z _________________ test_convert_stream_with_hints[test_vector3] _________________
2025-03-29T04:51:40.3406178Z 
2025-03-29T04:51:40.3407481Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3408771Z 
2025-03-29T04:51:40.3409003Z     @pytest.mark.parametrize("test_vector", GENERAL_TEST_VECTORS)
2025-03-29T04:51:40.3409537Z     def test_convert_stream_with_hints(test_vector):
2025-03-29T04:51:40.3410033Z         """Test the conversion of a stream with full stream info."""
2025-03-29T04:51:40.3410502Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3410830Z     
2025-03-29T04:51:40.3411091Z         stream_info = StreamInfo(
2025-03-29T04:51:40.3411713Z             extension=os.path.splitext(test_vector.filename)[1],
2025-03-29T04:51:40.3412190Z             mimetype=test_vector.mimetype,
2025-03-29T04:51:40.3412590Z             charset=test_vector.charset,
2025-03-29T04:51:40.3412938Z         )
2025-03-29T04:51:40.3413175Z     
2025-03-29T04:51:40.3413625Z         with open(os.path.join(TEST_FILES_DIR, test_vector.filename), "rb") as stream:
2025-03-29T04:51:40.3414170Z >           result = markitdown.convert(
2025-03-29T04:51:40.3414627Z                 stream, stream_info=stream_info, url=test_vector.url
2025-03-29T04:51:40.3415041Z             )
2025-03-29T04:51:40.3415198Z 
2025-03-29T04:51:40.3415333Z tests/test_module_vectors.py:86: 
2025-03-29T04:51:40.3415733Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3416204Z src/markitdown/_markitdown.py:284: in convert
2025-03-29T04:51:40.3416890Z     return self.convert_stream(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3417461Z src/markitdown/_markitdown.py:372: in convert_stream
2025-03-29T04:51:40.3418043Z     return self._convert(file_stream=stream, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3418633Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3418929Z 
2025-03-29T04:51:40.3419172Z self = <markitdown._markitdown.MarkItDown object at 0x7f427ff89580>
2025-03-29T04:51:40.3420109Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3421712Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', charset=None, filename=None, local_path=None, url=None)]
2025-03-29T04:51:40.3423218Z kwargs = {}, res = None
2025-03-29T04:51:40.3423787Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f4284072ff0>]
2025-03-29T04:51:40.3425542Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f42841e0260>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f427ff5be60>, priority=0.0), ...]
2025-03-29T04:51:40.3427192Z cur_pos = 0
2025-03-29T04:51:40.3427790Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3428368Z 
2025-03-29T04:51:40.3428481Z     def _convert(
2025-03-29T04:51:40.3428935Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3429486Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3429895Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3430287Z     
2025-03-29T04:51:40.3430605Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3431105Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3431521Z     
2025-03-29T04:51:40.3431882Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3432596Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3433485Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3434297Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3434812Z     
2025-03-29T04:51:40.3435185Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3435673Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3436015Z     
2025-03-29T04:51:40.3436344Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3437018Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3437508Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3438182Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3438648Z                 assert (
2025-03-29T04:51:40.3438980Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3439500Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3439990Z     
2025-03-29T04:51:40.3440274Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3440655Z     
2025-03-29T04:51:40.3440937Z                 # Copy any additional global options
2025-03-29T04:51:40.3441422Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3441934Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3442311Z     
2025-03-29T04:51:40.3442654Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3443138Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3443515Z     
2025-03-29T04:51:40.3443862Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3444343Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3444694Z     
2025-03-29T04:51:40.3445065Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3445625Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3446036Z     
2025-03-29T04:51:40.3446336Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3446959Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3447353Z     
2025-03-29T04:51:40.3447601Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3447963Z                 if stream_info is not None:
2025-03-29T04:51:40.3448542Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3449001Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3449404Z     
2025-03-29T04:51:40.3449673Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3450078Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3450434Z     
2025-03-29T04:51:40.3450830Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3451338Z                 _accepts = False
2025-03-29T04:51:40.3451656Z                 try:
2025-03-29T04:51:40.3452069Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3452577Z                 except NotImplementedError:
2025-03-29T04:51:40.3452948Z                     pass
2025-03-29T04:51:40.3453227Z     
2025-03-29T04:51:40.3453572Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3454028Z                 assert (
2025-03-29T04:51:40.3454353Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3454919Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3455432Z     
2025-03-29T04:51:40.3455693Z                 # Attempt the conversion
2025-03-29T04:51:40.3456056Z                 if _accepts:
2025-03-29T04:51:40.3456361Z                     try:
2025-03-29T04:51:40.3456948Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3457433Z                     except Exception:
2025-03-29T04:51:40.3457808Z                         failed_attempts.append(
2025-03-29T04:51:40.3458213Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3458657Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3459078Z                             )
2025-03-29T04:51:40.3459390Z                         )
2025-03-29T04:51:40.3459674Z                     finally:
2025-03-29T04:51:40.3459997Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3460507Z     
2025-03-29T04:51:40.3460766Z                 if res is not None:
2025-03-29T04:51:40.3461119Z                     # Normalize the content
2025-03-29T04:51:40.3461507Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3461991Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3462449Z                     )
2025-03-29T04:51:40.3462843Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3463299Z                     return res
2025-03-29T04:51:40.3463603Z     
2025-03-29T04:51:40.3463925Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3464379Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3464832Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3465578Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3466458Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3467148Z 
2025-03-29T04:51:40.3467380Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3467971Z _______________ test_convert_stream_without_hints[test_vector3] ________________
2025-03-29T04:51:40.3468362Z 
2025-03-29T04:51:40.3469528Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3470818Z 
2025-03-29T04:51:40.3471057Z     @pytest.mark.parametrize("test_vector", GENERAL_TEST_VECTORS)
2025-03-29T04:51:40.3471594Z     def test_convert_stream_without_hints(test_vector):
2025-03-29T04:51:40.3472264Z         """Test the conversion of a stream with no stream info."""
2025-03-29T04:51:40.3472715Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3473045Z     
2025-03-29T04:51:40.3473464Z         with open(os.path.join(TEST_FILES_DIR, test_vector.filename), "rb") as stream:
2025-03-29T04:51:40.3474077Z >           result = markitdown.convert(stream, url=test_vector.url)
2025-03-29T04:51:40.3474421Z 
2025-03-29T04:51:40.3474556Z tests/test_module_vectors.py:101: 
2025-03-29T04:51:40.3474958Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3475409Z src/markitdown/_markitdown.py:284: in convert
2025-03-29T04:51:40.3475932Z     return self.convert_stream(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3476495Z src/markitdown/_markitdown.py:372: in convert_stream
2025-03-29T04:51:40.3477246Z     return self._convert(file_stream=stream, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3477856Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3478154Z 
2025-03-29T04:51:40.3478418Z self = <markitdown._markitdown.MarkItDown object at 0x7f42859856d0>
2025-03-29T04:51:40.3479369Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3481001Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', charset=None, filename=None, local_path=None, url=None)]
2025-03-29T04:51:40.3482171Z kwargs = {}, res = None
2025-03-29T04:51:40.3482744Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f42859deb10>]
2025-03-29T04:51:40.3484554Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f42859dc8c0>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f42859f0470>, priority=0.0), ...]
2025-03-29T04:51:40.3486070Z cur_pos = 0
2025-03-29T04:51:40.3486981Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3487574Z 
2025-03-29T04:51:40.3487692Z     def _convert(
2025-03-29T04:51:40.3488138Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3488680Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3489094Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3489483Z     
2025-03-29T04:51:40.3489785Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3490264Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3490669Z     
2025-03-29T04:51:40.3491029Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3491741Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3492629Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3493446Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3493951Z     
2025-03-29T04:51:40.3494311Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3494801Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3495143Z     
2025-03-29T04:51:40.3495475Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3495989Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3496471Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3497118Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3497554Z                 assert (
2025-03-29T04:51:40.3497878Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3498569Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3499027Z     
2025-03-29T04:51:40.3499318Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3499693Z     
2025-03-29T04:51:40.3499966Z                 # Copy any additional global options
2025-03-29T04:51:40.3500446Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3500936Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3501297Z     
2025-03-29T04:51:40.3501624Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3502111Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3502481Z     
2025-03-29T04:51:40.3502819Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3503300Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3503680Z     
2025-03-29T04:51:40.3504105Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3504658Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3505052Z     
2025-03-29T04:51:40.3505357Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3505852Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3506252Z     
2025-03-29T04:51:40.3506504Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3507044Z                 if stream_info is not None:
2025-03-29T04:51:40.3507441Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3507898Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3508313Z     
2025-03-29T04:51:40.3508583Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3508989Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3509355Z     
2025-03-29T04:51:40.3509752Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3510440Z                 _accepts = False
2025-03-29T04:51:40.3510770Z                 try:
2025-03-29T04:51:40.3511185Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3511709Z                 except NotImplementedError:
2025-03-29T04:51:40.3512079Z                     pass
2025-03-29T04:51:40.3512359Z     
2025-03-29T04:51:40.3512695Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3513132Z                 assert (
2025-03-29T04:51:40.3513455Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3513999Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3514522Z     
2025-03-29T04:51:40.3514783Z                 # Attempt the conversion
2025-03-29T04:51:40.3515152Z                 if _accepts:
2025-03-29T04:51:40.3515456Z                     try:
2025-03-29T04:51:40.3515856Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3516338Z                     except Exception:
2025-03-29T04:51:40.3516882Z                         failed_attempts.append(
2025-03-29T04:51:40.3517292Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3517725Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3518133Z                             )
2025-03-29T04:51:40.3518443Z                         )
2025-03-29T04:51:40.3518739Z                     finally:
2025-03-29T04:51:40.3519071Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3519431Z     
2025-03-29T04:51:40.3519672Z                 if res is not None:
2025-03-29T04:51:40.3520028Z                     # Normalize the content
2025-03-29T04:51:40.3520586Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3521079Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3521565Z                     )
2025-03-29T04:51:40.3521961Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3522429Z                     return res
2025-03-29T04:51:40.3522740Z     
2025-03-29T04:51:40.3523068Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3523526Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3523984Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3524703Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3525568Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3526048Z 
2025-03-29T04:51:40.3526266Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3527007Z _____________________ test_convert_file_uri[test_vector3] ______________________
2025-03-29T04:51:40.3527385Z 
2025-03-29T04:51:40.3528571Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3529872Z 
2025-03-29T04:51:40.3530126Z     @pytest.mark.parametrize("test_vector", GENERAL_TEST_VECTORS)
2025-03-29T04:51:40.3530627Z     def test_convert_file_uri(test_vector):
2025-03-29T04:51:40.3531039Z         """Test the conversion of a file:// URI."""
2025-03-29T04:51:40.3531435Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3531759Z     
2025-03-29T04:51:40.3532024Z >       result = markitdown.convert(
2025-03-29T04:51:40.3532518Z             Path(os.path.join(TEST_FILES_DIR, test_vector.filename)).as_uri(),
2025-03-29T04:51:40.3533022Z             url=test_vector.url,
2025-03-29T04:51:40.3533338Z         )
2025-03-29T04:51:40.3533491Z 
2025-03-29T04:51:40.3533625Z tests/test_module_vectors.py:134: 
2025-03-29T04:51:40.3534212Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3534680Z src/markitdown/_markitdown.py:269: in convert
2025-03-29T04:51:40.3535197Z     return self.convert_uri(source, stream_info=stream_info, **_kwargs)
2025-03-29T04:51:40.3535757Z src/markitdown/_markitdown.py:413: in convert_uri
2025-03-29T04:51:40.3536189Z     return self.convert_local(
2025-03-29T04:51:40.3536585Z src/markitdown/_markitdown.py:325: in convert_local
2025-03-29T04:51:40.3537307Z     return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3537891Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3538197Z 
2025-03-29T04:51:40.3538454Z self = <markitdown._markitdown.MarkItDown object at 0x7f4284001970>
2025-03-29T04:51:40.3539416Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3541396Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', c....pptx', local_path='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx', url=None)]
2025-03-29T04:51:40.3542912Z kwargs = {}, res = None
2025-03-29T04:51:40.3543481Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f427ff46540>]
2025-03-29T04:51:40.3545233Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f42847e2150>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f42859b6bd0>, priority=0.0), ...]
2025-03-29T04:51:40.3546882Z cur_pos = 0
2025-03-29T04:51:40.3547468Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3548218Z 
2025-03-29T04:51:40.3548330Z     def _convert(
2025-03-29T04:51:40.3548783Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3549336Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3549755Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3550149Z     
2025-03-29T04:51:40.3550457Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3550946Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3551366Z     
2025-03-29T04:51:40.3551732Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3552439Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3553312Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3554134Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3554646Z     
2025-03-29T04:51:40.3555030Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3555528Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3555862Z     
2025-03-29T04:51:40.3556193Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3556858Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3557347Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3557854Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3558294Z                 assert (
2025-03-29T04:51:40.3558622Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3559140Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3559634Z     
2025-03-29T04:51:40.3559923Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3560291Z     
2025-03-29T04:51:40.3560732Z                 # Copy any additional global options
2025-03-29T04:51:40.3561235Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3561740Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3562117Z     
2025-03-29T04:51:40.3562454Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3562935Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3563305Z     
2025-03-29T04:51:40.3563656Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3564146Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3564515Z     
2025-03-29T04:51:40.3564890Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3565457Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3565855Z     
2025-03-29T04:51:40.3566171Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3566788Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3567186Z     
2025-03-29T04:51:40.3567443Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3567806Z                 if stream_info is not None:
2025-03-29T04:51:40.3568215Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3568674Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3569088Z     
2025-03-29T04:51:40.3569357Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3569763Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3570128Z     
2025-03-29T04:51:40.3570525Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3571259Z                 _accepts = False
2025-03-29T04:51:40.3571564Z                 try:
2025-03-29T04:51:40.3571978Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3572491Z                 except NotImplementedError:
2025-03-29T04:51:40.3572858Z                     pass
2025-03-29T04:51:40.3573145Z     
2025-03-29T04:51:40.3573480Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3573929Z                 assert (
2025-03-29T04:51:40.3574254Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3574817Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3575344Z     
2025-03-29T04:51:40.3575601Z                 # Attempt the conversion
2025-03-29T04:51:40.3575967Z                 if _accepts:
2025-03-29T04:51:40.3576277Z                     try:
2025-03-29T04:51:40.3576838Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3577325Z                     except Exception:
2025-03-29T04:51:40.3577704Z                         failed_attempts.append(
2025-03-29T04:51:40.3578109Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3578542Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3578957Z                             )
2025-03-29T04:51:40.3579258Z                         )
2025-03-29T04:51:40.3579546Z                     finally:
2025-03-29T04:51:40.3579867Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3580223Z     
2025-03-29T04:51:40.3580467Z                 if res is not None:
2025-03-29T04:51:40.3580822Z                     # Normalize the content
2025-03-29T04:51:40.3581205Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3581700Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3582177Z                     )
2025-03-29T04:51:40.3582579Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3583218Z                     return res
2025-03-29T04:51:40.3583531Z     
2025-03-29T04:51:40.3583842Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3584284Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3584724Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3585450Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3586313Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3587251Z 
2025-03-29T04:51:40.3587541Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3588110Z _____________________ test_convert_data_uri[test_vector3] ______________________
2025-03-29T04:51:40.3588474Z 
2025-03-29T04:51:40.3589544Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati...ase of the caption is Human-written.](Picture4.jpg)'], must_not_include=['data:image/jpeg;base64,/9j/4AAQSkZJRgABAQE'])
2025-03-29T04:51:40.3590698Z 
2025-03-29T04:51:40.3590912Z     @pytest.mark.parametrize("test_vector", GENERAL_TEST_VECTORS)
2025-03-29T04:51:40.3591373Z     def test_convert_data_uri(test_vector):
2025-03-29T04:51:40.3591732Z         """Test the conversion of a data URI."""
2025-03-29T04:51:40.3592079Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3592363Z     
2025-03-29T04:51:40.3592569Z         data = ""
2025-03-29T04:51:40.3592962Z         with open(os.path.join(TEST_FILES_DIR, test_vector.filename), "rb") as stream:
2025-03-29T04:51:40.3593497Z             data = base64.b64encode(stream.read()).decode("utf-8")
2025-03-29T04:51:40.3593901Z         mimetype = test_vector.mimetype
2025-03-29T04:51:40.3594525Z         data_uri = f"data:{mimetype};base64,{data}"
2025-03-29T04:51:40.3594857Z     
2025-03-29T04:51:40.3595083Z >       result = markitdown.convert(
2025-03-29T04:51:40.3595392Z             data_uri,
2025-03-29T04:51:40.3595650Z             url=test_vector.url,
2025-03-29T04:51:40.3595927Z         )
2025-03-29T04:51:40.3596056Z 
2025-03-29T04:51:40.3596451Z tests/test_module_vectors.py:155: 
2025-03-29T04:51:40.3596958Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3597358Z src/markitdown/_markitdown.py:269: in convert
2025-03-29T04:51:40.3597811Z     return self.convert_uri(source, stream_info=stream_info, **_kwargs)
2025-03-29T04:51:40.3598286Z src/markitdown/_markitdown.py:431: in convert_uri
2025-03-29T04:51:40.3598642Z     return self.convert_stream(
2025-03-29T04:51:40.3598991Z src/markitdown/_markitdown.py:372: in convert_stream
2025-03-29T04:51:40.3599511Z     return self._convert(file_stream=stream, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3600031Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3600287Z 
2025-03-29T04:51:40.3600517Z self = <markitdown._markitdown.MarkItDown object at 0x7f4289181cd0>
2025-03-29T04:51:40.3600996Z file_stream = <_io.BytesIO object at 0x7f428922d850>
2025-03-29T04:51:40.3602076Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', charset=None, filename=None, local_path=None, url=None)]
2025-03-29T04:51:40.3603095Z kwargs = {}, res = None
2025-03-29T04:51:40.3603675Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f4289232690>]
2025-03-29T04:51:40.3605252Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f4289232cf0>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f4289233500>, priority=0.0), ...]
2025-03-29T04:51:40.3606573Z cur_pos = 0
2025-03-29T04:51:40.3607195Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3607832Z 
2025-03-29T04:51:40.3607933Z     def _convert(
2025-03-29T04:51:40.3608319Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3608805Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3609167Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3609513Z     
2025-03-29T04:51:40.3609780Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3610116Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3610386Z     
2025-03-29T04:51:40.3610634Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3611116Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3611706Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3612253Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3612592Z     
2025-03-29T04:51:40.3612839Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3613167Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3613389Z     
2025-03-29T04:51:40.3613613Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3613960Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3614281Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3614614Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3614912Z                 assert (
2025-03-29T04:51:40.3615129Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3615471Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3615908Z     
2025-03-29T04:51:40.3616104Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3616352Z     
2025-03-29T04:51:40.3616538Z                 # Copy any additional global options
2025-03-29T04:51:40.3616972Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3617300Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3617559Z     
2025-03-29T04:51:40.3617794Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3618125Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3618366Z     
2025-03-29T04:51:40.3618587Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3618910Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3619155Z     
2025-03-29T04:51:40.3619413Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3619785Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3620046Z     
2025-03-29T04:51:40.3620247Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3620565Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3620831Z     
2025-03-29T04:51:40.3621001Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3621241Z                 if stream_info is not None:
2025-03-29T04:51:40.3621510Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3621830Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3622134Z     
2025-03-29T04:51:40.3622317Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3622587Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3622820Z     
2025-03-29T04:51:40.3623098Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3623443Z                 _accepts = False
2025-03-29T04:51:40.3623779Z                 try:
2025-03-29T04:51:40.3624070Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3624420Z                 except NotImplementedError:
2025-03-29T04:51:40.3624667Z                     pass
2025-03-29T04:51:40.3624858Z     
2025-03-29T04:51:40.3625085Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3625385Z                 assert (
2025-03-29T04:51:40.3625607Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3625973Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3626318Z     
2025-03-29T04:51:40.3626492Z                 # Attempt the conversion
2025-03-29T04:51:40.3626845Z                 if _accepts:
2025-03-29T04:51:40.3627056Z                     try:
2025-03-29T04:51:40.3627318Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3627649Z                     except Exception:
2025-03-29T04:51:40.3627913Z                         failed_attempts.append(
2025-03-29T04:51:40.3628192Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3628491Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3628768Z                             )
2025-03-29T04:51:40.3628981Z                         )
2025-03-29T04:51:40.3629179Z                     finally:
2025-03-29T04:51:40.3629408Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3629648Z     
2025-03-29T04:51:40.3629812Z                 if res is not None:
2025-03-29T04:51:40.3630052Z                     # Normalize the content
2025-03-29T04:51:40.3630311Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3630766Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3631081Z                     )
2025-03-29T04:51:40.3631347Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3631659Z                     return res
2025-03-29T04:51:40.3631862Z     
2025-03-29T04:51:40.3632080Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3632381Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3632683Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3633157Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3633735Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3634056Z 
2025-03-29T04:51:40.3634207Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3634612Z __________________ test_convert_keep_data_uris[test_vector1] ___________________
2025-03-29T04:51:40.3634859Z 
2025-03-29T04:51:40.3635654Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati.../jpeg;base64,/9j/4AAQSkZJRgABAQE'], must_not_include=['![This phrase of the caption is Human-written.](Picture4.jpg)'])
2025-03-29T04:51:40.3636503Z 
2025-03-29T04:51:40.3636794Z     @pytest.mark.parametrize("test_vector", DATA_URI_TEST_VECTORS)
2025-03-29T04:51:40.3637144Z     def test_convert_keep_data_uris(test_vector):
2025-03-29T04:51:40.3637462Z         """Test API functionality when keep_data_uris is enabled"""
2025-03-29T04:51:40.3637769Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3637991Z     
2025-03-29T04:51:40.3638161Z         # Test local file conversion
2025-03-29T04:51:40.3638409Z >       result = markitdown.convert(
2025-03-29T04:51:40.3638690Z             os.path.join(TEST_FILES_DIR, test_vector.filename),
2025-03-29T04:51:40.3638991Z             keep_data_uris=True,
2025-03-29T04:51:40.3639216Z             url=test_vector.url,
2025-03-29T04:51:40.3639426Z         )
2025-03-29T04:51:40.3639519Z 
2025-03-29T04:51:40.3639739Z tests/test_module_vectors.py:171: 
2025-03-29T04:51:40.3640017Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3640317Z src/markitdown/_markitdown.py:271: in convert
2025-03-29T04:51:40.3640670Z     return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3641040Z src/markitdown/_markitdown.py:325: in convert_local
2025-03-29T04:51:40.3641421Z     return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3641791Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3641983Z 
2025-03-29T04:51:40.3642150Z self = <markitdown._markitdown.MarkItDown object at 0x7f42840704d0>
2025-03-29T04:51:40.3642765Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3644059Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', c....pptx', local_path='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx', url=None)]
2025-03-29T04:51:40.3645070Z kwargs = {'keep_data_uris': True}, res = None
2025-03-29T04:51:40.3645503Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f427ff44bf0>]
2025-03-29T04:51:40.3646747Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f427ff47c50>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f427ff47620>, priority=0.0), ...]
2025-03-29T04:51:40.3647701Z cur_pos = 0
2025-03-29T04:51:40.3648105Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3648629Z 
2025-03-29T04:51:40.3648700Z     def _convert(
2025-03-29T04:51:40.3649010Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3649383Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3649667Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3649931Z     
2025-03-29T04:51:40.3650136Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3650458Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3650732Z     
2025-03-29T04:51:40.3650973Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3651447Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3652024Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3652553Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3652887Z     
2025-03-29T04:51:40.3653131Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3653455Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3653677Z     
2025-03-29T04:51:40.3653889Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3654232Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3654559Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3654893Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3655192Z                 assert (
2025-03-29T04:51:40.3655419Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3655755Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3656077Z     
2025-03-29T04:51:40.3656275Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3656523Z     
2025-03-29T04:51:40.3656805Z                 # Copy any additional global options
2025-03-29T04:51:40.3657243Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3657579Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3657829Z     
2025-03-29T04:51:40.3658052Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3658370Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3658606Z     
2025-03-29T04:51:40.3658827Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3659144Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3659388Z     
2025-03-29T04:51:40.3659633Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3660002Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3660255Z     
2025-03-29T04:51:40.3660459Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3660779Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3661044Z     
2025-03-29T04:51:40.3661207Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3661443Z                 if stream_info is not None:
2025-03-29T04:51:40.3661712Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3662020Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3662295Z     
2025-03-29T04:51:40.3662477Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3662740Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3662987Z     
2025-03-29T04:51:40.3663255Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3663778Z                 _accepts = False
2025-03-29T04:51:40.3663992Z                 try:
2025-03-29T04:51:40.3664273Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3664620Z                 except NotImplementedError:
2025-03-29T04:51:40.3664870Z                     pass
2025-03-29T04:51:40.3665058Z     
2025-03-29T04:51:40.3665282Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3665577Z                 assert (
2025-03-29T04:51:40.3665785Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3666158Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3666509Z     
2025-03-29T04:51:40.3666777Z                 # Attempt the conversion
2025-03-29T04:51:40.3667020Z                 if _accepts:
2025-03-29T04:51:40.3667220Z                     try:
2025-03-29T04:51:40.3667497Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3667843Z                     except Exception:
2025-03-29T04:51:40.3668113Z                         failed_attempts.append(
2025-03-29T04:51:40.3668404Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3668703Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3668988Z                             )
2025-03-29T04:51:40.3669200Z                         )
2025-03-29T04:51:40.3669399Z                     finally:
2025-03-29T04:51:40.3669622Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3669857Z     
2025-03-29T04:51:40.3670026Z                 if res is not None:
2025-03-29T04:51:40.3670266Z                     # Normalize the content
2025-03-29T04:51:40.3670514Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3670841Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3671156Z                     )
2025-03-29T04:51:40.3671417Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3671725Z                     return res
2025-03-29T04:51:40.3672046Z     
2025-03-29T04:51:40.3672264Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3672562Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3672853Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3673333Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3673907Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3674224Z 
2025-03-29T04:51:40.3674380Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3674779Z _______________ test_convert_stream_keep_data_uris[test_vector1] _______________
2025-03-29T04:51:40.3675033Z 
2025-03-29T04:51:40.3675831Z test_vector = FileTestVector(filename='test.pptx', mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentati.../jpeg;base64,/9j/4AAQSkZJRgABAQE'], must_not_include=['![This phrase of the caption is Human-written.](Picture4.jpg)'])
2025-03-29T04:51:40.3676802Z 
2025-03-29T04:51:40.3676975Z     @pytest.mark.parametrize("test_vector", DATA_URI_TEST_VECTORS)
2025-03-29T04:51:40.3677333Z     def test_convert_stream_keep_data_uris(test_vector):
2025-03-29T04:51:40.3677666Z         """Test the conversion of a stream with no stream info."""
2025-03-29T04:51:40.3677970Z         markitdown = MarkItDown()
2025-03-29T04:51:40.3678188Z     
2025-03-29T04:51:40.3678371Z         stream_info = StreamInfo(
2025-03-29T04:51:40.3678646Z             extension=os.path.splitext(test_vector.filename)[1],
2025-03-29T04:51:40.3678959Z             mimetype=test_vector.mimetype,
2025-03-29T04:51:40.3679222Z             charset=test_vector.charset,
2025-03-29T04:51:40.3679456Z         )
2025-03-29T04:51:40.3679739Z     
2025-03-29T04:51:40.3680007Z         with open(os.path.join(TEST_FILES_DIR, test_vector.filename), "rb") as stream:
2025-03-29T04:51:40.3680373Z >           result = markitdown.convert(
2025-03-29T04:51:40.3680723Z                 stream, stream_info=stream_info, keep_data_uris=True, url=test_vector.url
2025-03-29T04:51:40.3681064Z             )
2025-03-29T04:51:40.3681166Z 
2025-03-29T04:51:40.3681266Z tests/test_module_vectors.py:195: 
2025-03-29T04:51:40.3681530Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3681824Z src/markitdown/_markitdown.py:284: in convert
2025-03-29T04:51:40.3682309Z     return self.convert_stream(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3682802Z src/markitdown/_markitdown.py:372: in convert_stream
2025-03-29T04:51:40.3683341Z     return self._convert(file_stream=stream, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3684004Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2025-03-29T04:51:40.3684288Z 
2025-03-29T04:51:40.3684486Z self = <markitdown._markitdown.MarkItDown object at 0x7f427f888b90>
2025-03-29T04:51:40.3685241Z file_stream = <_io.BufferedReader name='/home/runner/work/markitdown/markitdown/packages/markitdown/tests/test_files/test.pptx'>
2025-03-29T04:51:40.3686438Z stream_info_guesses = [StreamInfo(mimetype='application/vnd.openxmlformats-officedocument.presentationml.presentation', extension='.pptx', charset=None, filename=None, local_path=None, url=None)]
2025-03-29T04:51:40.3687617Z kwargs = {'keep_data_uris': True}, res = None
2025-03-29T04:51:40.3688206Z failed_attempts = [<markitdown._exceptions.FailedConversionAttempt object at 0x7f42892b06e0>]
2025-03-29T04:51:40.3689427Z sorted_registrations = [ConverterRegistration(converter=<markitdown.converters._epub_converter.EpubConverter object at 0x7f428927c830>, prior...tration(converter=<markitdown.converters._audio_converter.AudioConverter object at 0x7f428927d8b0>, priority=0.0), ...]
2025-03-29T04:51:40.3690481Z cur_pos = 0
2025-03-29T04:51:40.3691032Z stream_info = StreamInfo(mimetype=None, extension=None, charset=None, filename=None, local_path=None, url=None)
2025-03-29T04:51:40.3691608Z 
2025-03-29T04:51:40.3691719Z     def _convert(
2025-03-29T04:51:40.3692248Z         self, *, file_stream: BinaryIO, stream_info_guesses: List[StreamInfo], **kwargs
2025-03-29T04:51:40.3692799Z     ) -> DocumentConverterResult:
2025-03-29T04:51:40.3693180Z         res: Union[None, DocumentConverterResult] = None
2025-03-29T04:51:40.3693487Z     
2025-03-29T04:51:40.3693874Z         # Keep track of which converters throw exceptions
2025-03-29T04:51:40.3694306Z         failed_attempts: List[FailedConversionAttempt] = []
2025-03-29T04:51:40.3694725Z     
2025-03-29T04:51:40.3695082Z         # Create a copy of the page_converters list, sorted by priority.
2025-03-29T04:51:40.3695616Z         # We do this with each call to _convert because the priority of converters may change between calls.
2025-03-29T04:51:40.3708579Z         # The sort is guaranteed to be stable, so converters with the same priority will remain in the same order.
2025-03-29T04:51:40.3709393Z         sorted_registrations = sorted(self._converters, key=lambda x: x.priority)
2025-03-29T04:51:40.3709773Z     
2025-03-29T04:51:40.3710048Z         # Remember the initial stream position so that we can return to it
2025-03-29T04:51:40.3710395Z         cur_pos = file_stream.tell()
2025-03-29T04:51:40.3710632Z     
2025-03-29T04:51:40.3710858Z         for stream_info in stream_info_guesses + [StreamInfo()]:
2025-03-29T04:51:40.3711218Z             for converter_registration in sorted_registrations:
2025-03-29T04:51:40.3711556Z                 converter = converter_registration.converter
2025-03-29T04:51:40.3711907Z                 # Sanity check -- make sure the cur_pos is still the same
2025-03-29T04:51:40.3712223Z                 assert (
2025-03-29T04:51:40.3712448Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3712994Z                 ), f"File stream position should NOT change between guess iterations"
2025-03-29T04:51:40.3713321Z     
2025-03-29T04:51:40.3713529Z                 _kwargs = {k: v for k, v in kwargs.items()}
2025-03-29T04:51:40.3713787Z     
2025-03-29T04:51:40.3713975Z                 # Copy any additional global options
2025-03-29T04:51:40.3714310Z                 if "llm_client" not in _kwargs and self._llm_client is not None:
2025-03-29T04:51:40.3714647Z                     _kwargs["llm_client"] = self._llm_client
2025-03-29T04:51:40.3714896Z     
2025-03-29T04:51:40.3715124Z                 if "llm_model" not in _kwargs and self._llm_model is not None:
2025-03-29T04:51:40.3715490Z                     _kwargs["llm_model"] = self._llm_model
2025-03-29T04:51:40.3715728Z     
2025-03-29T04:51:40.3715957Z                 if "style_map" not in _kwargs and self._style_map is not None:
2025-03-29T04:51:40.3716280Z                     _kwargs["style_map"] = self._style_map
2025-03-29T04:51:40.3716524Z     
2025-03-29T04:51:40.3716915Z                 if "exiftool_path" not in _kwargs and self._exiftool_path is not None:
2025-03-29T04:51:40.3717280Z                     _kwargs["exiftool_path"] = self._exiftool_path
2025-03-29T04:51:40.3717543Z     
2025-03-29T04:51:40.3717750Z                 # Add the list of converters for nested processing
2025-03-29T04:51:40.3718069Z                 _kwargs["_parent_converters"] = self._converters
2025-03-29T04:51:40.3718338Z     
2025-03-29T04:51:40.3718499Z                 # Add legaxy kwargs
2025-03-29T04:51:40.3718744Z                 if stream_info is not None:
2025-03-29T04:51:40.3719016Z                     if stream_info.extension is not None:
2025-03-29T04:51:40.3719326Z                         _kwargs["file_extension"] = stream_info.extension
2025-03-29T04:51:40.3719601Z     
2025-03-29T04:51:40.3719780Z                     if stream_info.url is not None:
2025-03-29T04:51:40.3720059Z                         _kwargs["url"] = stream_info.url
2025-03-29T04:51:40.3720307Z     
2025-03-29T04:51:40.3720571Z                 # Check if the converter will accept the file, and if so, try to convert it
2025-03-29T04:51:40.3720921Z                 _accepts = False
2025-03-29T04:51:40.3721267Z                 try:
2025-03-29T04:51:40.3721554Z                     _accepts = converter.accepts(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3721904Z                 except NotImplementedError:
2025-03-29T04:51:40.3722183Z                     pass
2025-03-29T04:51:40.3722376Z     
2025-03-29T04:51:40.3722606Z                 # accept() should not have changed the file stream position
2025-03-29T04:51:40.3722902Z                 assert (
2025-03-29T04:51:40.3723119Z                     cur_pos == file_stream.tell()
2025-03-29T04:51:40.3723485Z                 ), f"{type(converter).__name__}.accept() should NOT change the file_stream position"
2025-03-29T04:51:40.3723831Z     
2025-03-29T04:51:40.3724003Z                 # Attempt the conversion
2025-03-29T04:51:40.3724246Z                 if _accepts:
2025-03-29T04:51:40.3724684Z                     try:
2025-03-29T04:51:40.3724988Z                         res = converter.convert(file_stream, stream_info, **_kwargs)
2025-03-29T04:51:40.3725324Z                     except Exception:
2025-03-29T04:51:40.3725579Z                         failed_attempts.append(
2025-03-29T04:51:40.3725855Z                             FailedConversionAttempt(
2025-03-29T04:51:40.3726154Z                                 converter=converter, exc_info=sys.exc_info()
2025-03-29T04:51:40.3726429Z                             )
2025-03-29T04:51:40.3726795Z                         )
2025-03-29T04:51:40.3727053Z                     finally:
2025-03-29T04:51:40.3727282Z                         file_stream.seek(cur_pos)
2025-03-29T04:51:40.3727518Z     
2025-03-29T04:51:40.3727679Z                 if res is not None:
2025-03-29T04:51:40.3727918Z                     # Normalize the content
2025-03-29T04:51:40.3728177Z                     res.text_content = "\n".join(
2025-03-29T04:51:40.3728652Z                         [line.rstrip() for line in re.split(r"\r?\n", res.text_content)]
2025-03-29T04:51:40.3728966Z                     )
2025-03-29T04:51:40.3729224Z                     res.text_content = re.sub(r"\n{3,}", "\n\n", res.text_content)
2025-03-29T04:51:40.3729535Z                     return res
2025-03-29T04:51:40.3729736Z     
2025-03-29T04:51:40.3729954Z         # If we got this far without success, report any exceptions
2025-03-29T04:51:40.3730251Z         if len(failed_attempts) > 0:
2025-03-29T04:51:40.3730553Z >           raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3731030Z E           markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3731609Z E            - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3731935Z 
2025-03-29T04:51:40.3732090Z src/markitdown/_markitdown.py:611: FileConversionException
2025-03-29T04:51:40.3732646Z =============================== warnings summary ===============================
2025-03-29T04:51:40.3733245Z ../../../../../.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170
2025-03-29T04:51:40.3734377Z   /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3735532Z     warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3735932Z 
2025-03-29T04:51:40.3736145Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2025-03-29T04:51:40.3736585Z =========================== short test summary info ============================
2025-03-29T04:51:40.3738011Z FAILED tests/test_cli_vectors.py::test_output_to_stdout[test_vector3] - AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3738272Z     warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3738373Z   Traceback (most recent call last):
2025-03-29T04:51:40.3743927Z     File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3744135Z     File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3744559Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3744648Z       main()
2025-03-29T04:51:40.3745043Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3745150Z       result = markitdown.convert(
2025-03-29T04:51:40.3745226Z                ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3745646Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3745836Z       return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3745934Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3746353Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3746562Z       return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3746815Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3747222Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3747404Z       raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3747847Z   markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3748078Z    - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3748155Z   
2025-03-29T04:51:40.3748230Z   
2025-03-29T04:51:40.3748306Z assert 1 == 0
2025-03-29T04:51:40.3749107Z  +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '/home/runner/work/markitdown/markitdown/packages/markitdown/test...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3750120Z FAILED tests/test_cli_vectors.py::test_output_to_file[test_vector3] - AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3750371Z     warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3750468Z   Traceback (most recent call last):
2025-03-29T04:51:40.3750641Z     File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3750792Z     File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3751188Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3751260Z       main()
2025-03-29T04:51:40.3751622Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3751713Z       result = markitdown.convert(
2025-03-29T04:51:40.3751798Z                ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3752185Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3752365Z       return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3752457Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3752987Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3753191Z       return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3753289Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3753692Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3753875Z       raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3754130Z   markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3754353Z    - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3754431Z   
2025-03-29T04:51:40.3754499Z   
2025-03-29T04:51:40.3754576Z assert 1 == 0
2025-03-29T04:51:40.3755308Z  +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '-o', '/tmp/pytest-of-runner/pytest-0/pytest_tmp0/test.pptx.outpu...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3756353Z FAILED tests/test_cli_vectors.py::test_input_from_stdin_without_hints[test_vector3] - AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3756589Z     warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3756801Z   Traceback (most recent call last):
2025-03-29T04:51:40.3756967Z     File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3757235Z     File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3757636Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3757715Z       main()
2025-03-29T04:51:40.3758080Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3758178Z       result = markitdown.convert(
2025-03-29T04:51:40.3758253Z                ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3758658Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3758832Z       return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3758924Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3759343Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3759546Z       return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3759643Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3760037Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3760213Z       raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3760468Z   markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3760696Z    - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3760764Z   
2025-03-29T04:51:40.3760835Z   
2025-03-29T04:51:40.3760910Z assert 1 == 0
2025-03-29T04:51:40.3761700Z  +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '/home/runner/work/markitdown/markitdown/packages/markitdown/test...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3762850Z FAILED tests/test_cli_vectors.py::test_output_to_file_with_data_uris[test_vector1] - AssertionError: CLI exited with error: /home/runner/.local/share/hatch/env/virtual/markitdown/H_uUSwZ6/hatch-test.py3.12/lib/python3.12/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
2025-03-29T04:51:40.3763100Z     warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
2025-03-29T04:51:40.3763190Z   Traceback (most recent call last):
2025-03-29T04:51:40.3763350Z     File "<frozen runpy>", line 198, in _run_module_as_main
2025-03-29T04:51:40.3763489Z     File "<frozen runpy>", line 88, in _run_code
2025-03-29T04:51:40.3763883Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 224, in <module>
2025-03-29T04:51:40.3763955Z       main()
2025-03-29T04:51:40.3764324Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/__main__.py", line 197, in main
2025-03-29T04:51:40.3764418Z       result = markitdown.convert(
2025-03-29T04:51:40.3764501Z                ^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3764900Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 271, in convert
2025-03-29T04:51:40.3765070Z       return self.convert_local(source, stream_info=stream_info, **kwargs)
2025-03-29T04:51:40.3765162Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3765578Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 325, in convert_local
2025-03-29T04:51:40.3765767Z       return self._convert(file_stream=fh, stream_info_guesses=guesses, **kwargs)
2025-03-29T04:51:40.3765861Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-03-29T04:51:40.3766341Z     File "/home/runner/work/markitdown/markitdown/packages/markitdown/src/markitdown/_markitdown.py", line 611, in _convert
2025-03-29T04:51:40.3766519Z       raise FileConversionException(attempts=failed_attempts)
2025-03-29T04:51:40.3767043Z   markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3767283Z    - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3767359Z   
2025-03-29T04:51:40.3767423Z   
2025-03-29T04:51:40.3767496Z assert 1 == 0
2025-03-29T04:51:40.3768234Z  +  where 1 = CompletedProcess(args=['python', '-m', 'markitdown', '--keep-data-uris', '-o', '/tmp/pytest-of-runner/pytest-0/pytest_...on failed after 1 attempts:\n - PptxConverter threw RecursionError with message: maximum recursion depth exceeded\n\n').returncode
2025-03-29T04:51:40.3768716Z FAILED tests/test_module_vectors.py::test_convert_local[test_vector3] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3768949Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3769442Z FAILED tests/test_module_vectors.py::test_convert_stream_with_hints[test_vector3] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3769664Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3770163Z FAILED tests/test_module_vectors.py::test_convert_stream_without_hints[test_vector3] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3770382Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3770852Z FAILED tests/test_module_vectors.py::test_convert_file_uri[test_vector3] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3771069Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3771663Z FAILED tests/test_module_vectors.py::test_convert_data_uri[test_vector3] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3771886Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3772387Z FAILED tests/test_module_vectors.py::test_convert_keep_data_uris[test_vector1] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3772606Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3773108Z FAILED tests/test_module_vectors.py::test_convert_stream_keep_data_uris[test_vector1] - markitdown._exceptions.FileConversionException: File conversion failed after 1 attempts:
2025-03-29T04:51:40.3773314Z  - PptxConverter threw RecursionError with message: maximum recursion depth exceeded
2025-03-29T04:51:40.3773490Z ======= 11 failed, 130 passed, 31 skipped, 1 warning in 69.25s (0:01:09) =======
2025-03-29T04:51:40.6398663Z ##[error]Process completed with exit code 1.
2025-03-29T04:51:40.6499891Z Post job cleanup.
2025-03-29T04:51:40.7424006Z [command]/usr/bin/git version
2025-03-29T04:51:40.7465253Z git version 2.49.0
2025-03-29T04:51:40.7509336Z Temporarily overriding HOME='/home/runner/work/_temp/e9d92183-3a86-4bcb-8a34-b2a82404109b' before making global git config changes
2025-03-29T04:51:40.7510669Z Adding repository directory to the temporary git global config as a safe directory
2025-03-29T04:51:40.7515111Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/markitdown/markitdown
2025-03-29T04:51:40.7550506Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2025-03-29T04:51:40.7583766Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2025-03-29T04:51:40.7818032Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2025-03-29T04:51:40.7838683Z http.https://github.com/.extraheader
2025-03-29T04:51:40.7851417Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2025-03-29T04:51:40.7882741Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2025-03-29T04:51:40.8213475Z Cleaning up orphan processes
